### ISSUE TITLE
OverflowError: Python int too large to convert to C long for certain pdfs

### ISSUE THREAD

**Issue by RakshaBM** [pdfminer/pdfminer.six#177]
```
Traceback (most recent call last):
  File "/Users/rakshabm/Documents/Python/extract_from_pdf.py", line 95, in <module>
    interpreter.process_page(page)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/pdfinterp.py", line 852, in process_page
    self.render_contents(page.resources, page.contents, ctm=ctm)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/pdfinterp.py", line 862, in render_contents
    self.init_resources(resources)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/pdfinterp.py", line 362, in init_resources
    self.fontmap[fontid] = self.rsrcmgr.get_font(objid, spec)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/pdfinterp.py", line 200, in get_font
    font = PDFType3Font(self, spec)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/pdffont.py", line 622, in __init__
    PDFSimpleFont.__init__(self, descriptor, widths, spec)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/pdffont.py", line 547, in __init__
    self.cid2unicode = EncodingDB.get_encoding(name, diff)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/encodingdb.py", line 61, in get_encoding
    cid2unicode[cid] = name2unicode(x.name)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pdfminer/encodingdb.py", line 21, in name2unicode
    return six.unichr(int(m.group(0)))
OverflowError: Python int too large to convert to C long
```
The error is thrown for only certain pdfs

**Comment by ganeshtata** [comment]
Could you please upload the problematic PDFs? If not, what size are each of these PDFs?

**Comment by jtlz2** [comment]
Any advance on this front? :\

**Comment by pietermarsman** [comment]
A pull request was merged yesterday that changes the `name2unicode()` function. Not sure if it fixes this problem though since there is no example pdf here. 

You might try to pull pdfminer.six from master and test if for your pdf.

**Comment by pietermarsman** [comment]
I think this was only a problem in older versions of python (don't have any proof). In python3 it is possible to create very large ints. 

```
>>> int(math.factorial(200))
788657867364790503552363213932185062295135977687173263294742533244359449963403342920304284011984623904177212138919638830257642790242637105061926624952829931113462857270763317237396988943922445621451664240254033291864131227428294853277524242407573903240321257405579568660226031904170324062351700858796178922222789623703897374720000000000000000000000000000000000000000000000000L
```

**Comment by jtlz2** [comment]
@pietermarsman Perfect timing - thanks for the heads-up!

I just tried it (python 2.7.....) and it still threw the original error though :(

The workaround I had had was to put a try except around the return statement and on Exception rather replace with some pre-agreed character (with our sample PDF, which I'm afraid I cannot share because of GDPR etc., this only happens ~ 10 / 10,000 times). It would be a bit of a mission to patch pdfminer.six but it may be enough for now.

I hope when we migrate to python 3 by the end of the year this will all go away, but a stop-gap in the meantime would be amazing.

Thanks!

**Comment by pietermarsman** [comment]
Could you get an example of the input of `name2unicode()`?

**Comment by pietermarsman** [comment]
I'm going to work on #261 which is probably related and an example input that does work is perfect for a unittest :)

**Comment by jtlz2** [comment]
Of course! An example input is:

226215240241240240240240

That sounds amazing, please keep me posted! :)

**Comment by pietermarsman** [comment]
Do you also know what the output should be?

**Comment by jtlz2** [comment]
Unfortunately not :( I am not sure how to map it back. Can you still make progress?

**Comment by pietermarsman** [comment]
I've added a test and currently it fails because of the `OverflowError`. I am not sure if that will be fixed if I implement the Adobe Glyph List Specification correctly. We'll see.

**Comment by jtlz2** [comment]
OK I have a bit more on this:

If I cat the PDF-1.3 file there are some offending lines like the following one:

```
<</Differences [0 /226215240241240240240240 64 /space **[REDACTED]** /one /two /three /four /five /six /seven /eight /nine]/Type /Encoding>>endobj
```

So when I print out the offending character within encodingdb.py I get:

type(x) = <class 'pdfminer.psparser.PSLiteral'>
x = /'226215240241240240240240'
x.name = 226215240241240240240240

This is then parsed as that overflowing integer.

What is happening here? Are the slashes just missing on every third character?

It's possible it's a layer inserted by OCR and hence could contain some obscure characters.

What's the best way to proceed?

Thanks!

**Comment by pietermarsman** [comment]
I am not sure what those characters mean. They are certainly not in the Adobe glyph specification.

Splitting it into 3-digit numbers is probably intended here. Than it would represent 8 bytes, probably a unicode value. It would be nice to know which unicode character this is and if other pdf readers also show this character in the document.

In general I find it really hard to decide when to follow the spec for pdfs and when to be pragmatic and support correct conversion of actual pdfs.

**Comment by pietermarsman** [comment]
@ganeshtata This issue can be closed since it is fixed by #263
